### PR TITLE
Setting up ACME database automatically

### PR DESCRIPTION
**Pull request by edewata**
The PostgreSQL.setup() has been added to automatically create
the tables when the server initially connects to the database.
This eliminates the requirement to create the tables manually.
The docs have been updated accordingly.

### CODE REVIEW

File: base/acme/src/main/java/org/dogtagpki/acme/database/PostgreSQLDatabase.java [hunk 1]
```diff
@@ -110,6 +113,39 @@ public void connect() throws Exception {
...
+     * This method will create the tables if they do not exist.
+     */
+    public void setup() throws Exception {
+
+        logger.info("Setting up database");
+
+        String filename = "/usr/share/pki/acme/database/postgresql/create.sql";
+        String content = new String(Files.readAllBytes(Paths.get(filename)));
+
+        String[] statements = content.split(";");
```
Comment (cipherboy) [inline base/acme/src/main/java/org/dogtagpki/acme/database/PostgreSQLDatabase.java:126]: Yuck. I guess there's no other way of doing this with JDBC as far as I can tell. I think having it in the initialization script is as good as anything, so I guess we're stuck with the initialization and parsing logic here. I'd maybe take the alternative route of a separate CLI to setup databases in general (so this occurs once every time the admin executes it instead of once every start up), but I understand why this is attractive for containers.

File: base/common/src/org/dogtagpki/nss/NSSExtensionGenerator.java [hunk 2]
```diff
@@ -253,6 +254,16 @@ public AuthInfoAccessExtension createAIAExtension() throws Exception {
             extension.addAccessDescription(method, location);
         }
 
+        // Call AuthInfoAccessExtension.encode() to generate the extensionValue,
+        // otherwise the extensionValue will be null.
+        //
+        // TODO: Implement AuthInfoAccessExtension.getExtensionValue() to generate
+        // the extensionValue whenever it's needed.
+
+        try (ByteArrayOutputStream os = new ByteArrayOutputStream()) {
```
Comment (cipherboy) [inline base/common/src/org/dogtagpki/nss/NSSExtensionGenerator.java:263]: I'm fine keeping this change, but perhaps it was meant to be part of a separate PR?

Comment (edewata) [reply base/common/src/org/dogtagpki/nss/NSSExtensionGenerator.java:263]: Yeah, this is actually unrelated to ACME database, I just thought it's small enough to stick into this PR :)
The problem appears when I was testing the latest code. I'm not sure why it didn't appear before.

File: base/acme/src/main/java/org/dogtagpki/acme/database/PostgreSQLDatabase.java (hunk 1 above)
Comment (edewata) [reply base/acme/src/main/java/org/dogtagpki/acme/database/PostgreSQLDatabase.java:126]: I was hoping to find a SQL parser in PostgreSQL JDBC driver, but there's none, so the best solution I have now is splitting the SQL file like above.

Right now the `setup()` will create the initial tables, but in the future we might use it to upgrade the tables as well. So when the container image is updated, it will automatically upgrade the database too on restart.

**Comment by edewata** [comment]
@cipherboy Thanks for the ACK. I'll wait until tomorrow before merging in case there are other comments.

File: base/acme/src/main/java/org/dogtagpki/acme/database/PostgreSQLDatabase.java (hunk 1 above)
Comment (cipherboy) [reply base/acme/src/main/java/org/dogtagpki/acme/database/PostgreSQLDatabase.java:126]: Yeah that's fair. My one concern that I'd need to test is what if we include comments in `create.sql` later? Will this try and execute the comments? And there can't be a `;` in the comment... Hmmm...

Comment (edewata) [reply base/acme/src/main/java/org/dogtagpki/acme/database/PostgreSQLDatabase.java:126]: I tried adding these two comment styles and they actually work just fine:
```
-- comment
```
and
```
/* comment */
```
But yeah, they can't contain a `;` for now.
